src/corelibs/U2Formats/src/FastaFormat.cpp

```diff
@@ -321,15 +321,20 @@ static void saveSequenceObject(IOAdapterWriter &writer, const U2SequenceObject *
     writeHeaderToFile(writer, sequence->getSequenceName(), os);
     CHECK_OP(os, );
     qint64 sequenceLength = sequence->getSequenceLength();
-    for (int i = 0; i < sequenceLength; i += FastaFormat::FASTA_SEQUENCE_LINE_LENGTH) {
-        qint64 chunkSize = qMin((qint64)FastaFormat::FASTA_SEQUENCE_LINE_LENGTH, sequenceLength - i);
+    // Reading DBI line-by-line is very expensive. Read by bigger chunks and split them into lines in memory.
+    qint64 maxChunkSize = 1000 * FastaFormat::FASTA_SEQUENCE_LINE_LENGTH;
```

Comment:
What `1000` is?

Reply:
This is a constant that makes the original chunk size (FASTA_SEQUENCE_LINE_LENGTH) several order of magnitude bigger, so the expensive per-chunk transactional code overhead becomes negligible. This can be any big number, there are only 2 limitations here: it must be a multiple of FASTA_SEQUENCE_LINE_LENGTH and small enough to fit into memory.